cairo/ethereum/cancun/state.cairo

```diff
@@ -181,3 +188,106 @@ func get_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
 
     return value;
 }
+
+func set_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
+    address: Address, key: Bytes32, value: U256
+) {
+    alloc_locals;
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    // Assert that the account exists
+    let account = get_account_optional(address);
+    if (cast(account.value, felt) == 0) {
+        with_attr error_message("Cannot set storage on non-existent account") {
```

Comment:
Not sure if this should be an Exception of type AssertionError instead. I need to think more about it

Reply:
should I add todo?

Reply:
think about which cases lead to this, yes